lib/generators/super_good/solidus_taxjar/install_generator.rb

```diff
@@ -10,6 +10,9 @@ def create_initializer_file
             Spree.config do |config|
               config.tax_calculator_class = SuperGood::SolidusTaxjar::TaxCalculator
             end
+            Spree::Event.subscribe 'shipment_shipped' do |shipment|
+              SuperGood::SolidusTaxjar::Reporting.report_transaction(shipment.order)
+            end
```

Comment:
Is there a reason why we are doing this in an initializer. The standard way is to introduce a subscriber class which can contain all our reporting callbacks https://guides.solidus.io/developers/events/overview.html.

Reply:
To expand on Chris's comment: I do think this code needs to be within the extension and not in the app, for extension maintainability reasons.

- We may change the `#report_transaction` method signature someday.
- This may cause errors in the host application if they remove the `solidus_taxjar` extension someday.

Reply:
@Noah-Silvera can correct me on this, but I think it's to give the users of the extension the option of not opting-in to our new reporting behaviour. If we do this within the extension, it would cause a problem for users who've implemented their own reporting solutions?

Reply:
I would argue that if an extension user wanted to implement their own reporting solution they should set `enable_reporting` to `false and then create their own `shipment_shipped` subscriber manually.

I think that our reporting implementation should always work the same way if `enable_reporting` is set to `true`. And I think that means that the logic should live inside of our extension. I worry that asking end users to commit generated code may be confusing and lead to hard-to-debug issues being reported.